# Q&A Activity Question Week -06

Read the data set into python



1.	Find out how many countries are there in the dataset.

2.	Find out how many countries with respect to the region are available in the dataset

3.	On average which region has the highest
    
    a) GDP per capita (current US$)
    
    b)	International trade: Imports (million US$)

4.	Find out which region has more consistent with respect to GDP per capita (current US$)

#### Importing required libraries

In [1]:
import numpy as np
import pandas as pd

#### Reading the data set into python

Note: Both dataset and python notebook are in same directory

In [2]:
data=pd.read_excel("Dataset_w6.xlsx")

Checking random sample of dataset

In [3]:
data.sample(5)

country          Region  Surface area (km2)  \
194      Zambia   EasternAfrica              752612   
98       Latvia  NorthernEurope               64573   
120  Mozambique   EasternAfrica              799380   
95       Kuwait     WesternAsia               17818   
134      Panama  CentralAmerica               75320   

     Population in thousands (2017)  GDP per capita (current US$)  \
194                           17094                        1311.1   
98                             1950                       13704.1   
120                           29669                         529.2   
95                             4136                       29303.9   
134                            4099                       13268.1   

     Agricultural production index   \
194                             179   
98                              130   
120                             137   
95                              199   
134                             117   

     International trade: Imports (million US$)  
194                                        7442  
98                                        13596  
120                                        5295  
95                                        30825  
134                                       11697

### 1. Find out how many countries are there in the dataset.

In [4]:
count1=data['country'].count()
print("No of countries in the data set: " , count1)

No of countries in the data set:  196


In [5]:
Num=len(data['country'])
print("No of countries in the data set: " , Num)

No of countries in the data set:  196


Note: The count() method and len() returns the number of elements in the column country in the dataset

### 2. Find out how many countries with respect to the region are available in the dataset

In [6]:
print('No. of countries with respect to region:')
data.groupby("Region", as_index=False).country.count()

No. of countries with respect to region:


Region  country
0           Caribbean       16
1      CentralAmerica        8
2         CentralAsia        5
3       EasternAfrica       17
4         EasternAsia        7
5       EasternEurope       10
6           Melanesia        5
7          Micronesia        4
8        MiddleAfrica        9
9      NorthernAfrica        5
10    NorthernAmerica        4
11     NorthernEurope       10
12            Oceania        2
13          Polynesia        5
14  South-easternAsia       11
15       SouthAmerica       12
16     SouthernAfrica        5
17       SouthernAsia        9
18     SouthernEurope       12
19      WesternAfrica       15
20        WesternAsia       18
21      WesternEurope        7

Note: groupby() function split data into separate regions to perform computations

### 3.	On average which region has the highest

####  a) GDP per capita (current USD)

In [7]:
Avg1=data.groupby("Region")[['GDP per capita (current US$)']].mean()
a=Avg1.sort_values(by=['GDP per capita (current US$)'],ascending=False)
a.head(1)

GDP per capita (current US$)
Region                                       
NorthernAmerica                       57659.0

#### Insight: Northern America region has the highest GDP per capita - 57659.0 USD

Note: The sort_values() function is used to sort by the values along either axis. Default is along index

####    b)	International trade: Imports (million USD)

In [8]:
x=data.groupby('Region')['International trade: Imports (million US$)'].mean()
x.nlargest(1)

Region
NorthernAmerica    663552.25
Name: International trade: Imports (million US$), dtype: float64

#### Insight: Northern America region has the highest average of International trade: 663552.25 million USD

Note: The nlargest() function is used to get the first n rows ordered by columns in descending order

### 4. Find out which region has more consistent with respect to GDP per capita (current US$)

In [9]:
std1=data.groupby('Region')[['GDP per capita (current US$)']].std()
a=std1.sort_values(by=['GDP per capita (current US$)'])
a.head(1)

GDP per capita (current US$)
Region                                     
WesternAfrica                    815.095399

Note: Standard Deviation (SD) is measured as the spread of data distribution in the given data set.A low SD indicates that the data points tend to be close to the mean.

####  Insights: Western African region GDP is more consistent 

### Participants
#### 1. Reshma Joseph
#### 2. Sreehari T
#### 3. Akshitha M K
#### 4. Sneha P Saji